sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -66,6 +65,27 @@ internal class AmqpMessageConverter
             return BuildAmqpMessageFromEvent(source, partitionKey);
         }
 
+        /// <summary>
+        ///   Converts a given set of <see cref="EventData" /> instances into a batch AMQP representation.
+        /// </summary>
+        ///
+        /// <param name="source">The set of source events to convert.</param>
+        /// <param name="partitionKey">The partition key to associate with the batch, if any.</param>
+        ///
+        /// <returns>The AMQP message that represents a batch containing the converted set of event data.</returns>
+        ///
+        /// <remarks>
+        ///   The caller is assumed to hold ownership over the message once it has been created, including
+        ///   ensuring proper disposal.
+        /// </remarks>
+        ///
+        public virtual AmqpMessage CreateBatchFromEvents(List<EventData> source,
```

Comment:
Added an overload because we need to keep the existing version with `IEnumerable<EventData>` signature to handle the producer `SendAsync` call that takes an enumerable.

Reply:
Not entirely sure this is needed. Pushed a few updates to https://github.com/Azure/azure-sdk-for-net/pull/26911 that seem to indicate they are not but I might have missed something since I'm in weekend mode 😪

Reply:
Going down the path of switching to List entirely, I realized that might not be the best strategy. So the comments in the other PR

Reply:
Based on this, I took another look and realized that we're doing a transform in the `EventHubProducerClient` to materialize the enumerable already.   Rather than keep this as `IEnumerable` all the way down, I swapped over to `IReadOnlyCollection` and switched over to `ToArray` to materialize things.    

That let me eliminate the overload and lock down to `IReadOnlyCollection` in the message converter.   The only hitch is needing to pull the first element off of the collection if its the only one present.   I ended up having to pattern match in a `switch` to use the indexer for the common cases of `List<EventData>` and `EventData[]` to avoid allocating the enumerator otherwise.